tests/e2e/vmlistener_test.go

```diff
@@ -186,6 +192,102 @@ func (s *VMListenerTestParams) TestFailoverAcrossVM(c *C) {
 	misc.LogTestEnd(c.TestName())
 }
 
+// TestVolumeAttachedForHostdRestart - verify an in use volume
+// remains attached when hostd is restarted.
+// 1. Restart hostd
+// 2. Verify verify volume stays as attached
+// 3. Kill the vm
+// 4. Status of the volume should be detached.
+func (s *VMListenerTestParams) TestVolumeAttachedForHostdRestart(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// 1. Restart hostd
+	out, err := ssh.InvokeCommand(s.esx, restartHostd)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Give hostd time to start up
+	misc.SleepForSec(30)
```

Comment:
I'm not clear of the details of the two failures that you ran into above. But we have been using polling in other scenarios and it's working perfectly fine.

"If hostd isn't ready thats something the ESX service needs to take care of." - that's exactly the point. If "hostd status" tells us the status is up and running, but hostd is not ready to start accepting requests, then it's hostd bug that ESX service should take care of.

Sleeping for a fixed amount of time is not robust because the time might be very different depending on different setup (vmfs vs. vsan, local vm vs. nimbus vm, etc.)